src/vs/workbench/contrib/codeEditor/browser/editorLineNumberMenu.ts

```diff
@@ -62,21 +62,22 @@ export class EditorLineNumberContextMenu extends Disposable implements IEditorCo
 	}
 
 	public show(e: IEditorMouseEvent) {
-		// on macOS ctrl+click is interpreted as right click
-		if (!e.event.rightButton && !(isMacintosh && e.event.leftButton && e.event.ctrlKey)) {
-			return;
-		}
-
-		const menu = this.menuService.createMenu(MenuId.EditorLineNumberContext, this.contextKeyService);
-
 		const model = this.editor.getModel();
-		if (!e.target.position || !model || e.target.type !== MouseTargetType.GUTTER_LINE_NUMBERS && e.target.type !== MouseTargetType.GUTTER_GLYPH_MARGIN) {
+
+		// on macOS ctrl+click is interpreted as right click
+		if (!e.event.rightButton && !(isMacintosh && e.event.leftButton && e.event.ctrlKey)
+			|| e.target.type !== MouseTargetType.GUTTER_LINE_NUMBERS && e.target.type !== MouseTargetType.GUTTER_GLYPH_MARGIN
+			|| !e.target.position || !model
+		) {
 			return;
 		}
 
 		const anchor = { x: e.event.posx, y: e.event.posy };
 		const lineNumber = e.target.position.lineNumber;
 
+		const contextKeyService = this.contextKeyService.createOverlay([['editorLineNumber', lineNumber]]);
```

Comment:
I was wondering about that, but every other consumer of `createOverlay` doesn't dispose it.

Reply:
Maybe worth an issue for the other cases then--and I'll update this PR to dispose it.

Reply:
It looks like dispose no-ops on it, so probably okay.

Reply:
I think the better thing would splitting out the IContextKeyService interface so that .dispose() doesn't have to be defined on something that doesn't need to be dispose.